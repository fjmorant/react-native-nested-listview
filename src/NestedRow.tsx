/* @flow */

import * as React from 'react'
import isEqual from 'react-fast-compare'
import {StyleSheet, View} from 'react-native'

const styles = StyleSheet.create({
    nestedRow: {
        flex: 1,
        justifyContent: 'center',
    },
})

export interface IProps {
    height?: number
    children: any
    level?: number
    paddingLeftIncrement?: number
    style?: any
}

const NestedRow = React.memo(
    ({
        height,
        children,
        level = 0,
        paddingLeftIncrement = 10,
        style,
    }: IProps) => (
        <View
            style={[
                styles.nestedRow,
                {
                    ...style,
                    paddingLeft: level * paddingLeftIncrement,
                },
                height ? {height} : {},
            ]}
        >
            {children}
        </View>
    ),
    isEqual
)

export default NestedRow
