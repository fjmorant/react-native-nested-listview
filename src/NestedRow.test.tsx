import React from 'react';
import { Text } from 'react-native';
import { NestedRow } from './NestedRow';
import { render, waitFor } from '@testing-library/react-native';

describe('NestedListView', () => {
  test('renders with succesfully', async () => {
    const { getByText } = render(
      <NestedRow level={1} style={{ borderColor: 'black', borderWidth: 1 }}>
        <Text>Test</Text>
      </NestedRow>,
    );

    await waitFor(() => {
      const component = getByText('Test');
      expect(component).toBeTruthy();
    });
  });

  test('renders without level passed', async () => {
    const { getByText } = render(
      <NestedRow style={{ borderColor: 'black', borderWidth: 1 }}>
        <Text>Test</Text>
      </NestedRow>,
    );

    await waitFor(() => {
      const component = getByText('Test');
      expect(component).toBeTruthy();
    });
  });

  test('renders with height passed', async () => {
    const { getByText } = render(
      <NestedRow
        level={1}
        height={60}
        paddingLeftIncrement={15}
        style={{ borderColor: 'black', borderWidth: 1 }}>
        <Text>Test</Text>
      </NestedRow>,
    );

    await waitFor(() => {
      const component = getByText('Test');
      expect(component).toBeTruthy();
    });
  });
});
