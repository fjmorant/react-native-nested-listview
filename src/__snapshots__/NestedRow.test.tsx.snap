// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedListView renders customised 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "paddingLeft": 15,
      },
      Object {
        "height": 60,
      },
    ]
  }
>
  <Text>
    Test
  </Text>
</View>
`;

exports[`NestedListView renders with simple array 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "paddingLeft": 10,
      },
      Object {},
    ]
  }
>
  <Text>
    Test
  </Text>
</View>
`;

exports[`NestedListView renders with simple array without level 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "paddingLeft": 0,
      },
      Object {},
    ]
  }
>
  <Text>
    Test
  </Text>
</View>
`;
